aristanetworks/ansible-avd
# Fix(eos_validate_state): Fix the DHCP not recognized error for STUN and Connectivity tests

## Change Summary

Fix the DHCP not recognized error for STUN and Connectivity tests. If the interface is dhcp then AvdTestStun and AvdTestP2PIPReachability tests try to collect the ip address from interface cause below error.
```
TASK [arista.avd.eos_validate_state : Run eos_validate_state_runner] *********************************************************************************************************************************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ValueError: 'dhcp' does not appear to be an IPv4 or IPv6 interface
fatal: [Edge20]: FAILED! => {"changed": false, "msg": "Error during plugin execution: 'dhcp' does not appear to be an IPv4 or IPv6 interface"}
```

## Related Issue(s)

Fixes #4725 

## Component(s) name

`arista.avd.eos_validate_state`

## Proposed changes
Fix the DHCP not recognized error for STUN and Connectivity tests. If the interface is dhcp then AvdTestStun and AvdTestP2PIPReachability tests try to collect the ip address from interface cause below error.

Added the logic to check if interface is dhcp interface or not to avoid error.

## How to test
Make the interface as DHCP and run the molecule. It should skip the test with below warning:
```
Host 'host1' interface 'Ethernet2' IP address is unavailable. AvdTestP2PIPReachability is skipped.
Host 'host1' interface 'Ethernet4' IP address is unavailable. AvdTestStun is skipped.
```

## Checklist

### User Checklist


- N/A

### Repository Checklist
